tests/crypto/array_initialization_test.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019 BitShares Blockchain Foundation, and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <boost/test/unit_test.hpp>
+
+#include <fc/crypto/elliptic.hpp>
+#include <fc/crypto/rand.hpp>
+
+#include <string.h>
+
+static void check_null_key()
+{
+   fc::ecc::public_key_data key1;
+   fc::ecc::public_key_data key2;
+   unsigned char zeroes[33];
+   static_assert( key1.size() == sizeof(zeroes), "Wrong array size!" );
```

Comment:
This does not compile with MSVC.

Reply:
@crypto-ape does `bitshares-core 3.3.2 release` compile with MSVC?

Reply:
To be clear, it does not compile with the MSVC bundled with VS 2015, it might possibly compile with the VS 2019, but I don't have the toolchain up and working.

I can test the 3.3.2 build with MSVS 2015 and will let you know.

Reply:
It looks like the `3.3.2` release compiles, had some problems with cURL not linking correctly, but other than that it compiled OK.

I have tried to compile the `develop` branch and there were some other problems related to the `template<>` already mentioned, but I could not reproduce the problem with the `static_assert`. 

Maybe I had some submodules misconfigured, but now it seems to work.